lib/Sema/TypeCheckDecl.cpp

```diff
@@ -3140,6 +3149,8 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
     }
 
     TC.checkParameterAttributes(FD->getParameters());
+
+    validateExplicitIntroVersion(FD);
```

Comment:
You're inserting a lot of these calls. Is there somewhere else you could put them? Maybe `checkAccessControl`? (Don't take this as "definitely do it this way"; I don't know if there are calls to `checkAccessControl` that wouldn't want this check. Just seems like a code smell.)

Reply:
I would like to have less call sites but I have yet to find another check that targets the same decls. Is there another file for this kind of check?